docs/blog/2025-06-20-pinned-zmk.md

```diff
@@ -0,0 +1,110 @@
+
+<!-- truncate -->
+
+## ZMK Versioning - The 'what' and the 'why'
+
+Around 6 months ago, ZMK began using versioning. Prior to this point, all ZMK users ran their firmware off of `main`. This meant that users would always get the latest and greatest changes whenever they built new firmware, but on occasion there would be breaking changes in `main`, resulting in plenty of user confusion.
+
+Avoiding this user confusion became a limiting factor when it came to making changes. [Zephyr](https://www.zephyrproject.org/), the RTOS that ZMK is based off of, made some breaking changes in version 3.6. Hence ZMK has been stuck on Zephyr 3.5 for over 1.5 years, while Zephyr has already moved on to version 4.1. Asking when ZMK will upgrade is a frequent question we get asked, as the new version comes with plenty of improvements and drivers for newer, exciting hardware.
+
+With versioning now in place, users can pin their ZMK configuration to a particular version. This prevents sudden breaking changes from happening without user opt-in, and frees ZMK contributors to make breaking changes more frequently, as is necessary.
+
+### ZMK's Versioning System
+
+ZMK follows the [Semantic Versioning](https://semver.org/) standard. Versions are `vX.Y.Z`. ZMK uses release branches and tags: Every major or minor version bump, a new branch and tag are created. The tag points to the latest patch version. For example, when releasing `v0.2.0` the branch `v0.2-branch` and tag `v0.2` were created. When patches are made, such as for `v0.2.1`, the branch and tag are updated to point to `v0.2.1` instead of `v0.2.0`. This way, users can receive non-breaking security and bugfixes without needing to bump their version.
+
+Whenever ZMK releases a major version with breaking changes, a proper changelog will be provided detailing the major changes from the previous version, along with clear instructions on how to upgrade and fix any breaking changes.
+
+## To Pin or Not To Pin
+
+If you are a casual user of ZMK, I would always recommend pinning your version. Unpinned ZMK configurations are mainly suitable for those more heavily involved or interested in ZMK development. Moving forwards, if you are on the `main` branch of ZMK you will be considered a (highly valued) tester, rather than a user.
+
+To ensure that you still update your ZMK version regularly and get the latest and greatest features that interest you, I would recommend setting up email notifications about new releases. You can do so on the GitHub repository:
+
+![Image of where to find the custom notifications](assets/2025-06-20-pinned-zmk/release1.png)
+![Image of setting custom notification](assets/2025-06-20-pinned-zmk/release2.png)
+
+## How To Pin Your ZMK Version
+
+:::info
+If you created your ZMK config on or after `21-06-2025`, then you are already pinned to what was the most recent ZMK version at the time of creation.
```

Comment:
```suggestion
If you created your ZMK config on or after `2025-06-21`, then you are already pinned to what was the most recent ZMK version at the time of creation.
```
I don't think DD-MM-YYYY is particularly common with hyphens, but in any case, using ISO8601 probably makes sense considering our international audience.